[Doc] Update canAccess doc (#9396)
marmelab/react-admin: docs/canAccess.md

## `action`

> If you want to check a specific action, for instance if the user can `delete` a `post` , you can specify this action in the dedicated prop: 

Reviewer: ```suggestion
If you want to check a specific action, for instance if the user can `delete` a `post`, you can specify this action in the dedicated prop: 
```

```tsx
const permissions = [
    { resource: 'posts', action: ['read', 'edit', 'create', 'delete'] },
];
canAccess({ permissions, resource: 'posts', action: 'read' }); // true
canAccess({ permissions, resource: 'posts', action: 'edit' }); // true
canAccess({ permissions, resource: 'posts', action: 'create' }); // true
canAccess({ permissions, resource: 'posts', action: 'delete' }); // true
canAccess({ permissions, resource: 'posts', action: 'export' }); // false
```

If you just want to know whether users can access any of the resources, you don't have to provide an `action`. For instance, here's how you may display different components depending on resources access rights in the dashboard:

```tsx

> import { Admin } from 'react-admin';

Reviewer: ```suggestion
import { Admin, usePermissions } from 'react-admin';
```

        ]),
};

const AccessDashboard = () => {
    const { permissions } = usePermissions();
    return (
        <>
            {canAccess({
                permissions,
                resource: 'commands',
            }) ? (
                <>List of last orders...</> // no access to this component
            ) : null}

            {canAccess({
                permissions,
                resource: 'products',
            }) ? (
               <>List of last products...</>
            ) : null}
            {canAccess({
                permissions,
                resource: 'categories',
            }) ? (
                <>List of last categories...</>
            ) : null}
        </>
    );
};

export const MyApp = () => (
    <Admin authProvider={authProvider} dataProvider={dataProvider} dashboard={AccessDashboard}>
        {/*...*/}
    </Admin>
);
```

In this example, users will see the list of last products and the list of last categories but they won't be able to see the list of last orders.

> **Note**: that [ra-rbac's `<Resource>` component](./AuthRBAC.md#resource) check automatically the `list`, `show `, `create` and `edit` actions, so you don't actually need to use `canAccess` if you want to restrict a whole resource.

Reviewer: ```suggestion
**Note**: [ra-rbac's `<Resource>` component](./AuthRBAC.md#resource) automatically checks for the `list`, `show`, `create` and `edit` actions, so you don't actually need to use `canAccess` if you want to restrict a whole resource.
```
